### PR TITLE
Migrate Config API module into JVM types/values

### PR DESCRIPTION
**Pull request by chamil321**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java [hunk 1]
```diff
@@ -92,4 +95,35 @@ private BMap buildBMap(Map<String, Object> section) {
...
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static MapValue buildMapValue(Map<String, Object> section) {
+        MapValue map = new MapValueImpl<String, Object>();
+
+        section.forEach((key, val) -> {
+            if (val instanceof String || val instanceof Long || val instanceof Double || val instanceof Boolean) {
+                map.put(key, val);
```
Comment (Kishanthan) [inline stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: What about other ref types?

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/SetConfig.java [hunk 2]
```diff
@@ -65,4 +66,8 @@ public void execute(Context context) {
 
         context.setReturnValues();
     }
+
+    public static void setConfig(Strand strand, String configKey, Object configValue) {
```
Comment (pubudu91) [inline stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/SetConfig.java]: Again, what's the purpose of this method? As with the contains implementation, a `Strand` is passed as a param but not made use of.

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java (hunk 1 above)
Comment (pubudu91) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: For configs, we only consider value types + maps of value types.

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/SetConfig.java (hunk 2 above)
Comment (Kishanthan) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/SetConfig.java]: Strand is a first argument to any methods in ballerina. The impl may or may not use the strand argument based on the requirement. For example, if you want to perform a non-blocking invocation within the impl, then you may use the strand ref.

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/Contains.java [hunk 3]
```diff
@@ -47,4 +48,8 @@ public void execute(Context context) {
         boolean containsKey = ConfigRegistry.getInstance().contains(configKey);
         context.setReturnValues(new BBoolean(containsKey));
     }
+
+    public static boolean contains(Strand strand, String configKey) {
```
[the remark this answers is not in the document]
Comment (vinok88) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/Contains.java]: That's how we model functions in ballerina. Strand should be the first param, regardless of its usage. Users may use strand to yield the execution etc..

**Pull request merged**